Add D202 pydocstyle check

pcandoalmeida opened the pull request:
Enabled D202 Pydocstyle check

* Fixed all pre-commit errors

Part of issue: #10742

---

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

pcandoalmeida committed:
Add D202 pydocstyle check

pcandoalmeida committed:
Add D202 pydocstyle check

mik-laj commented on airflow/providers/amazon/aws/hooks/glue_catalog.py:
```diff
@@ -104,12 +104,10 @@ def get_table(self, database_name, table_name):
         :type table_name: str
         :rtype: dict
-
         >>> hook = AwsGlueCatalogHook()
```
```
        """
        Get the information of the table

		.. code-block:: py

            hook = AwsGlueCatalogHook()
            r = hook.get_table('db', 'table_foo')
            r['Name'] = 'table_foo'

        :param database_name: Name of hive database (schema) @table belongs to
        :type database_name: str
        :param table_name: Name of hive table
        :type table_name: str
        :rtype: dict
        """
```
Can you try it?

pcandoalmeida replied:
Yep, that works, thanks @mik-laj!

pcandoalmeida committed:
Fix doc build error

pcandoalmeida committed:
Fix doc build error

kaxil reviewed (approved):
LGTM. Good work @pcandoalmeida

pcandoalmeida committed:
Merge branch 'master' into pydocstyle-D202

pcandoalmeida committed:
Merge latest Apache Airflow

pcandoalmeida committed:
Fix merge conflict

potiuk committed:
Separates out user documentation for production images. (#10998)

We have now much better user-facing documentation.

Only the parts interesting for users of the image are
separated out to the "docs" of Airflow.
The README and IMAGES.rst contains links to those
docs and internal details of the images respectively.

Fixes #10997.

potiuk committed:
All versions in CI yamls are not hard-coded any more (#10959)

GitHub Actions allow to use `fromJson` method to read arrays
or even more complex json objects into the CI workflow yaml files.

This, connected with set::output commands, allows to read the
list of allowed versions as well as default ones from the
environment variables configured in
./scripts/ci/libraries/initialization.sh

This means that we can have one plece in which versions are
configured. We also need to do it in "breeze-complete" as this is
a standalone script that should not source anything we added
BATS tests to verify if the versions in breeze-complete
correspond with those defined in the initialization.sh

Also we do not limit tests any more in regular PRs now - we run
all combinations of available versions. Our tests run quite a
bit faster now so we should be able to run more complete
matrixes. We can still exclude individual values of the matrixes
if this is too much.

MySQL 8 is disabled from breeze for now. I plan a separate follow
up PR where we will run MySQL 8 tests (they were not run so far)

potiuk committed:
Add Workflow to delete old artifacts (#11064)

XD-DENG committed:
[Doc] Correct description for macro task_instance_key_str (#11062)

Correction based on code https://github.com/apache/airflow/blob/master/airflow/models/taskinstance.py

dimberman committed:
Revert "KubernetesJobWatcher no longer inherits from Process (#11017)" (#11065)

This reverts commit 1539bd051cfbc41c1c7aa317fc7df82dab28f9f8.

flvndh committed:
Add JSON schema validation for Helm values (#10664)

fixes #10634

kaxil committed:
Get Airflow configs with sensitive data from CloudSecretManagerBackend (#11024)

kaxil committed:
Add some tasks using BashOperator in TaskGroup example dag (#11072)

Previously all the tasks in airflow/example_dags/example_task_group.py were using DummyOperator which does not go to executor and is marked as success in Scheduler itself so it would be good to have some tasks that aren't dummy operator to properly test TaskGroup functionality

kaxil committed:
Replace Airflow Slack Invite old link to short link (#11071)

Follow up to https://github.com/apache/airflow/pull/10034

https://apache-airflow-slack.herokuapp.com/ to https://s.apache.org/airflow-slack/

turbaszek committed:
Fix s.apache.org Slack link (#11078)

Remove ending / from s.apache.org Slack link

pcandoalmeida committed:
Merge latest Airflow
